### PR TITLE
Java: Add VS Code model editor queries

### PR DESCRIPTION
**Pull request by koesie10**
This adds two queries that support the CodeQL Model Editor feature in the CodeQL VS Code extension. These queries will be used to retrieve models that can be modeled using MaD. The query results they produce can also be converted to SARIF, hence the static strings in the query.

The `ModelEditor.qll` file's behavior is very similar to [`ExternalApi.qll`](https://github.com/github/codeql/blob/cd5973764b37d8720d1d4d31c31d8e0fc2d8a2c9/java/ql/src/Telemetry/ExternalApi.qll), but they are used for completely different purposes, and changes to the `ExternalApi.qll` do not necessarily need to propagate to `ModelEditor.qll`.

These queries have been in use for a few weeks now in the VS Code extension (see [`java.ts`](https://github.com/github/vscode-codeql/blob/93fc90fcd081f8d049440f3575e1bbbe74ff2db7/extensions/ql-vscode/src/model-editor/queries/java.ts)). Using this, we have confirmed that these queries return the results we need. Once these queries are merged and released, we will switch to resolving the queries from the query packs. We will do so based on the tags.

### CODE REVIEW

File: java/ql/src/utils/modeleditor/ApplicationModeEndpoints.ql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @name Fetch endpoints for use in the model editor (application mode)
+ * @description A list of 3rd party endpoints (methods) used in the codebase. Excludes test and generated code.
+ * @kind problem
+ * @problem.severity recommendation
```
Comment (aschackmull) [inline java/ql/src/utils/modeleditor/ApplicationModeEndpoints.ql:5]: That's almost certainly incorrect meta-data. I think you can just use `@kind table` and remove the `@problem.severity` line.

File: java/ql/src/utils/modeleditor/FrameworkModeEndpoints.ql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/**
+ * @name Fetch endpoints for use in the model editor (framework mode)
+ * @description A list of endpoints accessible (methods) for consumers of the library. Excludes test and generated code.
+ * @kind problem
+ * @problem.severity recommendation
```
Comment (aschackmull) [inline java/ql/src/utils/modeleditor/FrameworkModeEndpoints.ql:5]: same

**Comment by aschackmull** [comment]
The metadata should be fixed. Otherwise this looks fine, I think.

**Review by aschackmull — APPROVED** [review]

**Comment by koesie10** [comment]
@aschackmull Thanks for the review! I've made one last change to make it in line with the C# query that splits the `apiName` column into 4 separate columns. This reduces the number of string concatenations and should reduce stress on the string pool. It doesn't change any of the other behavior.

**Review by aschackmull — APPROVED** [review]

**Comment by koesie10** [comment]
I can [successfully run DCA](https://github.com/github/codeql-dca-main/issues/16422) against these queries and [the timings](https://github.com/github/codeql-dca-main/blob/data/koesie10/pr-14199-ced95e__nightly__Trivial/reports/summaries/csv/Meta_queries_analysis_time_per_source_(v0).csv#meta-queries-analysis-time-per-source-v0) look to be reasonable for most projects. I've created a PR to add a new query suite to DCA: https://github.com/github/codeql-dca/pull/1673

**Pull request merged**